truncate.cc

```diff
@@ -0,0 +1,78 @@
+#include <cstdlib>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <iomanip>
+#include <iterator>
+#include <random>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <cctype>
+#define a 0.8
+struct Point {
+  double x,y,r;
+  //std::vector<double> y;
+};
+
+int main(int argc, char** argv) {
+  std::string inputfile, outputfile;
+  if (argc == 3) {
+    inputfile = argv[1];
+    outputfile = argv[2];
+  } else {
+    std::cerr << "Incorret number of arguments." << std::endl;
+    std::exit(EXIT_FAILURE);
+  }
+  std::ifstream infile(inputfile);
+  std::ofstream ofs(outputfile);
+  std::vector<Point> points;
+  std::string line;
+ 
+  std::vector<std::string>vec;
+  if (infile.is_open()) {
+    while (std::getline(infile, line))
+      if(line!=""){
+      
+        std::string linesub = line.substr (0,50);
+        //   std::istringstream istream(line);
+        vec.emplace_back(linesub);
+      }
+    }
+  
+  infile.close();
+ 
+   double max_x = 0.;
+  for (auto it =vec.begin();it!=vec.end();it++)
+  {
+    std::istringstream pt(*it);
+    std::string s;
+    int pam = 0;
```

Comment:
what are these?

could you do look at https://github.com/cb-geo/mpm/blob/c1f0566a8c7a7eac4315738506ccf1c0c92053fb/include/read_mesh_ascii.tcc#L37

Reply:
I let the pointer *it to iterate every element in the vector, the elements are separated by the blank space. And then pass every element to string s. Maybe I take a detour to realize the function but I hope it makes sense.